### ISSUE TITLE
inconsistent treatment of hyperbolic functions with simplify

### ISSUE THREAD

**Issue by terben** [sympy/sympy#26592]
Here is my Ipython interaction with SymPy:

```
In [1]: import sympy as sp

In [2]: x, y = sp.symbols('x, y', real=True)

In [3]: sp.simplify(sp.sinh(x)**2 + 1)
Out[3]: cosh(x)**2

In [4]: sp.simplify(sp.sinh(x + y)**2 + 1)
Out[4]: cosh(x + y)**2

In [5]: sp.simplify(sp.cosh(x)**2 - 1)
Out[5]: sinh(x)**2

In [6]: sp.simplify(sp.cosh(x + y)**2 - 1)
Out[6]: cosh(x + y)**2 - 1

```

The results of the first three `In[]`- expressions meet my expectations but I would expect the Output of the last expression to yield `sp.sinh(x + y)**2`. 

```
In [7]: sp.__version__
Out[7]: '1.12'
```

**Comment by smichr** [comment]
Given that [4] works, it *is* curious that [6] does not.

**Comment by terben** [comment]
> Given that [4] works, it _is_ curious that [6] does not.

Yes! I am currently doing quite a lot of calculations involving hyperbolic functions and it is the third issue of this kind. Unfortunately, I do not know anything about SymPys internal implementation but perhaps #26593, #26579 and this issue are connected in some way (missing or errorneous implementaion of cosh, sinh properties).
